--- a/pyconfig.h	2015-03-13 13:11:55.000000000 +0100
+++ b/pyconfig.h	2015-03-13 13:00:15.000000000 +0100
@@ -309,14 +309,14 @@
 #define HAVE_GAMMA 1
 
 /* Define if we can use x64 gcc inline assembler */
-/* #undef HAVE_GCC_ASM_FOR_X64 */
+#define HAVE_GCC_ASM_FOR_X64 1
 
 /* Define if we can use gcc inline assembler to get and set x87 control word
    */
 #define HAVE_GCC_ASM_FOR_X87 1
 
 /* Define if your compiler provides __uint128_t */
-/* #undef HAVE_GCC_UINT128_T */
+#define HAVE_GCC_UINT128_T 1
 
 /* Define if you have the getaddrinfo function. */
 #define HAVE_GETADDRINFO 1
@@ -1344,7 +1344,7 @@
 #endif
 
 /* Define if arithmetic is subject to x87-style double rounding issue */
-#define X87_DOUBLE_ROUNDING 1
+/* #undef X87_DOUBLE_ROUNDING */
 
 /* Define on OpenBSD to activate all library features */
 /* #undef _BSD_SOURCE */
