From 151ef8fff20b676f7bb3636970cd92bd8d40d7f4 Mon Sep 17 00:00:00 2001
From: heyang <heyang@weiboyi.com>
Date: Mon, 20 May 2024 15:56:59 +0800
Subject: [PATCH] pool-frida

---
 gum/gum.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gum/gum.c b/gum/gum.c
index 239c70d1..77f143fd 100644
--- a/gum/gum.c
+++ b/gum/gum.c
@@ -306,7 +306,7 @@ gum_init_embedded (void)
   g_log_set_default_handler (gum_on_log_message, NULL);
   gum_do_init ();
 
-  g_set_prgname ("frida");
+  g_set_prgname ("ggbond");
 
 #if defined (HAVE_LINUX) && defined (HAVE_GLIBC)
   gum_libdl_prevent_unload ();
-- 
2.39.2 (Apple Git-143)

