From a49d09e49128bed89d0578bcee0d43dce86210e7 Mon Sep 17 00:00:00 2001
From: heyang <heyang@weiboyi.com>
Date: Wed, 22 May 2024 10:49:49 +0800
Subject: [PATCH 02/10] re_frida_server

---
 server/server.vala | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/server/server.vala b/server/server.vala
index 525c145e..7faa7476 100644
--- a/server/server.vala
+++ b/server/server.vala
@@ -1,7 +1,7 @@
 namespace Frida.Server {
 	private static Application application;
 
-	private const string DEFAULT_DIRECTORY = "re.frida.server";
+	private const string DEFAULT_DIRECTORY = null;
 	private static bool output_version = false;
 	private static string? listen_address = null;
 	private static string? certpath = null;
@@ -50,6 +50,7 @@ namespace Frida.Server {
 	};
 
 	private static int main (string[] args) {
+		DEFAULT_DIRECTORY = GLib.Uuid.string_random();
 		Environment.init ();
 
 #if DARWIN
-- 
2.39.2 (Apple Git-143)

