From 904cf40be13b4ed9b64a8f3a3413b28d8dc42195 Mon Sep 17 00:00:00 2001
From: heyang <heyang@weiboyi.com>
Date: Wed, 22 May 2024 11:00:35 +0800
Subject: [PATCH 06/10] embed_agent_2

---
 src/embed-agent.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/embed-agent.py b/src/embed-agent.py
index e08134e9..422efe31 100644
--- a/src/embed-agent.py
+++ b/src/embed-agent.py
@@ -97,6 +97,10 @@ def main(argv):
             random_name = "".join(random.sample("abcdefghijklmn", 11))
             print(f"[*] Patch `gum-js-loop` to `{random_name}`")
             os.system(f"sed -b -i s/gum-js-loop/{random_name}/g {embedded_agent}")    
+            # gmain thread
+            random_name = "".join(random.sample("abcdefghijklmn", 5))
+            print(f"[*] Patch `gmain` to `{random_name}`")
+            os.system(f"sed -b -i s/gmain/{random_name}/g {embedded_agent}")
             embedded_assets += [embedded_agent]
     elif host_os in {"freebsd", "qnx"}:
         embedded_agent = priv_dir / "frida-agent.so"
-- 
2.39.2 (Apple Git-143)

